### PR TITLE
Storage refactoring

### PR DESCRIPTION
**Pull request by Tinsane**
Refactored all storages out of WAL-G to decompose project structure.

### CODE REVIEW

File: internal/storages/s3/folder.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	AccessKeyIdSetting       = "AWS_ACCESS_KEY_ID"
+	SecretAccessKeySetting   = "AWS_SECRET_ACCESS_KEY"
+	SseSetting               = "S3_SSE"
+	SseKmsIdSetting          = "S3_SSE_KMS_ID"
+	StorageClassSetting      = "S3_STORAGE_CLASS"
+	UploadConcurrencySetting = "UPLOAD_CONCURRENCY"
+)
+
+// MaxRetries limit upload and download retries during interaction with S3
+var MaxRetries = 15
```
Comment (x4m) [inline internal/storages/s3/folder.go]: Keeping this magic const looks strange

File: test/lzo_test.go [hunk 2]
```diff
@@ -2,127 +2,118 @@
...
-	// extract.ExtractAll(np, out)
-
-	buf := &tools.BufferTarInterpreter{}
-	err := internal.ExtractAll(buf, out)
-	if err != nil {
-		b.Log(err)
-	}
-
-}
+//
```
Comment (x4m) [inline test/lzo_test.go]: Why these tests are discarded?

Comment (Tinsane) [reply test/lzo_test.go]: No

File: internal/storages/s3/folder.go (hunk 1 above)
Comment (Tinsane) [reply internal/storages/s3/folder.go]: Okay

Comment (x4m) [reply internal/storages/s3/folder.go]: Let's add this setting in a another PR? To keep scope of this clear.

**Pull request merged**